runtime/src/vote_account.rs

```diff
@@ -65,12 +65,13 @@ impl VoteAccount {
         self.0.account.owner()
     }
 
-    pub fn vote_state(&self) -> &Result<VoteState, Error> {
+    pub fn vote_state(&self) -> Result<&VoteState, &Error> {
```

Comment:
I guess the tradeoff is that this return type is ugly/weird.
Is the `.as_ref()` at call-sites really a big deal?

Reply:
i would argue that `&Result<T, E>` is uglier than `Result<&T, &E>`. we prefer `Option<&T>` to `&Option<T>` in most cases elsewhere to avoid the eye-twitch-inducing `&None`.  ofc, this would all be better if it were `Result<&T, E>`, but i don't think we can get to that without jumping through hoops